sql/rpl_gtid.cc

```diff
@@ -4029,3 +4065,18 @@ my_bool Intersecting_gtid_event_filter::has_finished()
   }
   return FALSE;
 }
+
+bool Intersecting_gtid_event_filter::verify_final_state()
+{
+  bool is_any_not_final= false;
+  Gtid_event_filter *subfilter;
+  for (size_t i= 0; i < m_filters.elements; ++i)
+  {
+    subfilter=
+      *reinterpret_cast<Gtid_event_filter **>(dynamic_array_ptr(&m_filters, i));
```

Comment:
I usually use
```suggestion
      *static_cast<Gtid_event_filter **>(dynamic_array_ptr(&m_filters, i));
```
, but `dynamic_array_ptr()` appearently returns `uchar*` rather than `void*`.